ang/test/civicase/case/services/add-case.service.spec.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-env jasmine */
+
+(($, loadForm, getCrmUrl) => {
+  describe('AddCaseService', () => {
+    let $window, AddCase, mockedFormPopUp, CaseCategoryWebformSettings;
+
+    beforeEach(() => {
+      $window = { location: { href: '' } };
+    });
+
+    describe('Button Visibility', () => {
+      let isButtonVisible;
+
+      beforeEach(module('civicase-base', 'civicase'));
+      beforeEach(() => {
+        injectDependencies();
+      });
+
+      describe('when the user can add new cases', () => {
+        beforeEach(() => {
+          CRM.checkPerm.and.returnValue(true);
+
+          isButtonVisible = AddCase.isVisible();
+        });
+
+        it('displays the button', () => {
+          expect(isButtonVisible).toBe(true);
+        });
+      });
+
+      describe('when the user cannot add new cases', () => {
+        beforeEach(() => {
+          CRM.checkPerm.and.returnValue(false);
+
+          isButtonVisible = AddCase.isVisible();
+        });
+
+        it('does not display the button', () => {
+          expect(isButtonVisible).toBe(false);
+        });
+      });
+    });
+
+    describe('click handler', () => {
+      beforeEach(module('civicase', ($provide) => {
+        CaseCategoryWebformSettings = jasmine.createSpyObj('CaseCategoryWebformSettings', ['getSettingsFor']);
+        $provide.value('CaseCategoryWebformSettings', CaseCategoryWebformSettings);
+      }));
+
+      describe('when the new case web form url configuration value is defined', () => {
+        beforeEach(module('civicase', ($provide) => {
+          $provide.value('$window', $window);
+          CaseCategoryWebformSettings.getSettingsFor.and.returnValue({
+            newCaseWebformUrl: '/someurl',
+            newCaseWebformClient: 'cid'
+          });
```

Comment:
Lets provide these mock services at the top of the file. The reason is that it's very distracting seeing these values provided multiple times. If we need to set them up in a certain way we can set the `returnValue` in a plain `beforeEach` without using `module` or `$provide` (since the mock service was already provided)